common/containers/Tabs/SendTransaction/index.jsx

```diff
@@ -439,50 +426,49 @@ export class SendTransaction extends React.Component {
         value = tokenBalance.balance.toString();
       }
     }
-
     let token = this.props.tokens.find(x => x.symbol === unit);
-
     this.setState({
       value,
       unit,
       token
     });
   };
 
-  generateTx = async () => {
-    const { nodeLib, wallet } = this.props;
-    const { token } = this.state;
-    const stateTxInfo = await this.getTransactionInfoFromState();
-
+  generateTxFromState = async () => {
+    const { nodeLib, wallet, gasPrice, network } = this.props;
+    const { token, unit, value, to, data, gasLimit } = this.state;
+    const chainId = network.chainId;
+    const transactionInput = {
+      token,
+      unit,
+      value,
+      to,
+      data
+    };
     try {
-      const transaction = await generateTransaction(
-        nodeLib,
-        {
-          ...stateTxInfo,
-          gasLimit: this.state.gasLimit,
-          gasPrice: this.props.gasPrice,
-          chainId: this.props.network.chainId
-        },
+      const signedTx = await generateCompleteTransaction(
         wallet,
-        token
+        nodeLib,
+        gasPrice,
+        gasLimit,
+        chainId,
+        transactionInput
       );
-      this.setState({ transaction });
+      this.setState({ transaction: signedTx });
     } catch (err) {
       this.props.showNotification('danger', err.message, 5000);
     }
   };
 
   openTxModal = () => {
-    if (this.state.transaction) {
-      this.setState({ showTxConfirm: true });
-    }
+    this.setState({ showTxConfirm: true });
```

Comment:
More concise if you combined `openTxModal/hideConfirmTx` into `toggleTxModal`

Reply:
I don't disagree, but the boilerplate of having some kind of force boolean argument, and not being able to use it directly from event bindings (Because first arg would be `ev`) make for a lot of annoyances.